golem-worker-executor-base/src/durable_host/filesystem/types.rs

```diff
@@ -245,14 +292,52 @@ impl<Ctx: WorkerCtx> HostDescriptor for DurableWorkerCtx<Ctx> {
         new_path: String,
     ) -> Result<(), FsError> {
         record_host_function_call("filesystem::types::descriptor", "rename_at");
-        HostDescriptor::rename_at(
+
+        let source_dir = Descriptor::dir(self.table.get(&self_)?)?;
+        let source_dir = source_dir.dir.clone();
+        let target_dir = Descriptor::dir(self.table.get(&self_)?)?;
+        let target_dir = target_dir.dir.clone();
+
+        let result = HostDescriptor::rename_at(
             &mut self.as_wasi_view(),
             self_,
-            old_path,
+            old_path.clone(),
             new_descriptor,
-            new_path,
+            new_path.clone(),
         )
-        .await
+        .await;
+
+        let target_f = Arc::new(target_dir.open(new_path.clone())?);
+        self.durable_file_times(
+            target_f,
+            "filesystem::types::descriptor::rename_at/target-file",
+        )
+        .await?;
+
+        let source_path: PathBuf = old_path.into();
+        let source_parent = match source_path.parent() {
+            Some(p) => Arc::new(source_dir.open_dir(p)?),
+            None => source_dir,
+        };
+
+        let target_path: PathBuf = new_path.into();
+        let target_parent = match target_path.parent() {
+            Some(p) => Arc::new(target_dir.open_dir(p)?),
+            None => target_dir,
+        };
+
+        self.durable_file_times(
```

Comment:
Here (and in other similar functions) we basically create two separate op-log entries, each of them restoring the access/modification times on a single file system entry. The entries are having the `ImportedFunctionInvoked` type though which feels a bit hacky. 

Probably we should always create a single entry, with storing a map of file times instead of a single one.

@adamgfraser what do you think?

Reply:
Or even create a dedicated "filesystem access log entry"

Reply:
The alternative appoach of only persisting `stat` and `stat-at` resolves this question